src/pykeen/evaluation/evaluation_loop.py

```diff
@@ -169,12 +175,16 @@ def evaluate(
         batch_size = determine_maximum_batch_size(
             batch_size=batch_size, device=self.model.device, maximum_batch_size=len(self.dataset)
         )
+        # set upper limit for slice size
+        # TODO: if we knew the targets here, we could guess this better
+        slice_size = max(self.model.num_entities, self.model.num_relations)
```

Comment:
there are parts of the pipeline that actually try to pass `slice_size` into this function (I am doing an experiment with typeddicts locally, which would make this PR much messier, but I found this)

so maybe slice_size should be an optional[int] where this gets calculated if it's not passed directly (or does this blow up AMO?)

Reply:
https://github.com/pykeen/pykeen/pull/1543/commits/a7d92311f367c8a2050cf6398d400097f7c43a2f

> (or does this blow up AMO?)

The function that is wrapped by AMO needs to pass an `int`